src/payment/strategies/stripe-upe/stripe-upe.mock.ts

```diff
@@ -0,0 +1,106 @@
+import { OrderRequestBody } from '../../../order';
+import { PaymentInitializeOptions } from '../../payment-request-options';
+
+import { StripePaymentMethodType, StripeUPEClient } from './stripe-upe';
+
+const gatewayId = 'stripeupe';
+
+export function getStripeUPEJsMock(): StripeUPEClient {
+    return {
+        elements: jest.fn(() => {
+            return {
+                create: jest.fn(() => {
+                    return {
+                        mount: jest.fn(),
+                        unmount: jest.fn(),
+                    };
+                }),
+                getElement: jest.fn().mockReturnValue(null),
+            };
+        }),
+        confirmPayment: jest.fn(),
+    };
+}
+
+export function getFailingStripeUPEJsMock(): StripeUPEClient {
+    return {
+        elements: jest.fn(() => {
+            return {
```

Comment:
this can be shortened as:

```ts
elements: jest.fn(() => ({
  create: jest.fn(() => ({
    mount: jest.fn(() => {
      throw new Error();
    }),
    unmount: jest.fn(),
  }),
}),
```